packages/admin-ui/src/Tooltip/Tooltip.tsx

```diff
@@ -1,30 +1,65 @@
-"use client";
-
 import * as React from "react";
 import * as TooltipPrimitive from "@radix-ui/react-tooltip";
 
 import { cn } from "~/utils";
 
-const TooltipProvider = TooltipPrimitive.Provider;
+export const TooltipProvider = TooltipPrimitive.Provider;
 
-const Tooltip = TooltipPrimitive.Root;
+const TooltipRoot = TooltipPrimitive.Root;
 
 const TooltipTrigger = TooltipPrimitive.Trigger;
 
 const TooltipContent = React.forwardRef<
     React.ElementRef<typeof TooltipPrimitive.Content>,
     React.ComponentPropsWithoutRef<typeof TooltipPrimitive.Content>
->(({ className, sideOffset = 4, ...props }, ref) => (
+>(({ className, ...props }, ref) => (
     <TooltipPrimitive.Content
         ref={ref}
-        sideOffset={sideOffset}
+        sideOffset={4}
         className={cn(
-            "z-50 overflow-hidden rounded-md border bg-popover px-3 py-1.5 text-sm text-popover-foreground shadow-md animate-in fade-in-0 zoom-in-95 data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2",
+            "z-50 rounded-md bg-popover px-3 py-2 max-w-64 text-sm text-popover-foreground animate-in fade-in-0 zoom-in-95 data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2",
             className
         )}
         {...props}
     />
 ));
+
 TooltipContent.displayName = TooltipPrimitive.Content.displayName;
 
-export { Tooltip, TooltipTrigger, TooltipContent, TooltipProvider };
+const TooltipArrow = (props: TooltipPrimitive.TooltipArrowProps) => (
+    <TooltipPrimitive.Arrow {...props} width={12} height={6} className={"fill-popover"} />
+);
+
+TooltipArrow.displayName = TooltipPrimitive.Arrow.displayName;
+
+export interface TooltipProps
+    extends Omit<TooltipPrimitive.TooltipContentProps, "content" | "children"> {
+    align?: TooltipPrimitive.TooltipContentProps["align"];
+    content: React.ReactNode;
+    onOpenChange?: TooltipPrimitive.TooltipProps["onOpenChange"];
+    showArrow?: boolean;
+    side?: TooltipPrimitive.TooltipContentProps["side"];
+    trigger: React.ReactNode;
+}
+
+export const Tooltip = ({
+    align,
+    content,
+    onOpenChange,
+    showArrow = true,
+    side,
+    trigger,
+    ...props
+}: TooltipProps) => {
```

Comment:
I'm having some consistency-related questions in my head... Like, here, we're using `content` and `trigger` props, and then, internally, we're using `TooltipContent` and `TooltipTrigger` components. But then again, with the tabs component you showed me today, there we're actually using `content=[<Tab .../>, <Tab .../>, ...]`. In other words, we're importing extra components from the UI package.

Are we okay with this slight inconsistency?

After that little Slack chat we had, I've actually followed the Tab component approach with the Card component I'm working on. Maybe we can discuss this further via a call.

Reply:
Our approaches might differ based on how the components are structured.

For this particular case, passing the TooltipContent and TooltipTrigger components as props might be unnecessary since the Tooltip Root component will always have one content and one trigger.

This is different from components like Tabs or Accordion, which handle multiple children and offer more customization options.

I’m happy to chat about this on a call—there’s always a chance I could be off base here! 😊

Reply:
Yeah, makes sense. 

One of the advantages of allowing users to pass `TooltipContent` and `TooltipTrigger` components is that this way they also can pass extra props to them, like maybe `className` or something similar. Alternative is not to have them exposed, but do it the way you did it here, in which case we'd need to expose a prop or two, so that users can pass extra props to these two components. And then I agree with you... not sure how viable of a solution is to just start adding props for these kinds of cases.

A disadvantage of having users pass these components is that now they need to know about them / import them, and then also, I saw you're adding `asChild` on `TooltipTrigger`, which means we'd need to set that for them internally, so they don't have to deal with it.

All in all, still not sure... I was also thinking maybe, on the Tab component side, instead of passing `TabsTrigger` and `TabsContent`, we pass POJOs.. but I think that's not good, and I like more what you already have with Tabs component.